tests/core/policies/test_intent_ted_policy.py

```diff
@@ -0,0 +1,382 @@
+    MASK,
+    SENTENCE,
+    IDS,
+)
+from rasa.shared.nlu.constants import INTENT
+from rasa.utils.tensorflow import model_data_utils
+from tests.core.test_policies import train_trackers
+
+
+class TestIntentTEDPolicy(TestTEDPolicy):
+    def create_policy(
+        self, featurizer: Optional[TrackerFeaturizer], priority: int
+    ) -> IntentTEDPolicy:
+        return IntentTEDPolicy(featurizer=featurizer, priority=priority)
+
+    @pytest.fixture(scope="class")
+    def featurizer(self) -> TrackerFeaturizer:
+        featurizer = IntentMaxHistoryTrackerFeaturizer(
+            IntentTokenizerSingleStateFeaturizer(), max_history=self.max_history
+        )
+        return featurizer
+
+    @pytest.mark.skip
+    def test_normalization(
+        self,
+        trained_policy: IntentTEDPolicy,
+        tracker: DialogueStateTracker,
+        default_domain: Domain,
+        monkeypatch: MonkeyPatch,
+    ):
+        # No normalization is done for IntentTEDPolicy and
+        # hence this test is overridden to do nothing.
+        assert True
+
+    def test_label_data_assembly(
+        self, trained_policy: IntentTEDPolicy, default_domain: Domain
+    ):
+        interpreter = RegexInterpreter()
+
+        # Construct input data
+        encoded_all_labels = trained_policy.featurizer.state_featurizer.encode_all_labels(
+            default_domain, interpreter
+        )
+        attribute_data, _ = model_data_utils.convert_to_data_format(encoded_all_labels)
+
+        assembled_label_data = trained_policy._assemble_label_data(
+            attribute_data, default_domain
+        )
+        assembled_label_data_signature = assembled_label_data.get_signature()
+
+        assert list(assembled_label_data_signature.keys()) == [
+            f"{LABEL}_{INTENT}",
+            LABEL,
+        ]
+        assert assembled_label_data.num_examples == len(default_domain.intents)
+        assert list(assembled_label_data_signature[f"{LABEL}_{INTENT}"].keys()) == [
+            MASK,
+            SENTENCE,
+        ]
+        assert list(assembled_label_data_signature[LABEL].keys()) == [IDS]
+        assert assembled_label_data_signature[f"{LABEL}_{INTENT}"][SENTENCE][
+            0
+        ].units == len(default_domain.intents)
+
+    async def test_prepared_data_for_threshold_prediction(
+        self,
+        trained_policy: IntentTEDPolicy,
+        default_domain: Domain,
+        stories_path: Path,
+    ):
+        training_trackers = await train_trackers(
+            default_domain, stories_path, augmentation_factor=0
+        )
+        interpreter = RegexInterpreter()
+        training_model_data, _ = trained_policy._prepare_for_training(
+            training_trackers, default_domain, interpreter
+        )
+
+        data_for_prediction = trained_policy._prepare_data_for_prediction(
+            training_model_data
+        )
+
+        assert all(
+            [key in PREDICTION_FEATURES for key in data_for_prediction.data.keys()]
+        )
```

Comment:
I think you can write this as `data_for_prediction.data.keys() <= PREDICTION_FEATURES` or `data_for_prediction.data.keys().issubset(PREDICTION_FEATURES)`.